crates/ty_python_semantic/src/types.rs

```diff
@@ -1128,7 +1132,8 @@ impl<'db> Type<'db> {
             | Type::AlwaysTruthy
             | Type::PropertyInstance(_)
             // might inherit `Any`, but subtyping is still reflexive
-            | Type::ClassLiteral(_) => true,
+            | Type::ClassLiteral(_)
+            | Type::TypedDict(_) => true,
```

Comment:
is it true that subtyping is always reflexive for a `TypedDict` type? What about gradual `TypedDict` types? I don't think this `TypedDict` is a subtype of itself, because it's a gradual type:

```py
from typing import Any, TypedDict

class Foo(TypedDict):
    x: Any
```